cypress/integration/ete-okta/registration_2.6.cy.ts

```diff
@@ -683,4 +683,329 @@ describe('Registration flow - Split 2/2', () => {
+
+			cy.contains('Enter your code');
+			cy.contains(unregisteredEmail);
+			cy.contains('send again');
+			cy.contains('try another address');
+
+			cy.checkForEmailAndGetDetails(
+				unregisteredEmail,
+				timeRequestWasMade1,
+			).then(({ body, codes }) => {
+				// email
+				expect(body).to.have.string('Your verification code');
+				expect(codes?.length).to.eq(1);
+				const code = codes?.[0].value;
+				expect(code).to.match(/^\d{6}$/);
+
+				// passcode page
+				cy.url().should('include', '/register/email-sent');
+				const timeRequestWasMade2 = new Date();
+				cy.contains('send again').click();
+				cy.checkForEmailAndGetDetails(
+					unregisteredEmail,
+					timeRequestWasMade2,
+				).then(({ body, codes }) => {
+					// email
+					expect(body).to.have.string('Your verification code');
+					expect(codes?.length).to.eq(1);
+					const code = codes?.[0].value;
+					expect(code).to.match(/^\d{6}$/);
+
+					// passcode page
+					cy.url().should('include', '/register/email-sent');
+
+					cy.get('input[name=code]').type(code!);
+					cy.contains('Submit passcode').click();
+
+					cy.url().should('contain', '/welcome/password');
+				});
+			});
+		});
+
+		it('change email functionality', () => {
+			const unregisteredEmail = randomMailosaurEmail();
+			cy.visit(`/register/email?usePasscodeRegistration=true`);
+
+			const timeRequestWasMade = new Date();
+			cy.get('input[name=email]').type(unregisteredEmail);
+			cy.get('[data-cy="main-form-submit-button"]').click();
+
+			cy.contains('Enter your code');
+			cy.contains(unregisteredEmail);
+			cy.contains('send again');
+			cy.contains('try another address');
+
+			cy.checkForEmailAndGetDetails(unregisteredEmail, timeRequestWasMade).then(
+				({ body, codes }) => {
+					// email
+					expect(body).to.have.string('Your verification code');
+					expect(codes?.length).to.eq(1);
+					const code = codes?.[0].value;
+					expect(code).to.match(/^\d{6}$/);
+
+					// passcode page
+					cy.url().should('include', '/register/email-sent');
+					cy.contains('try another address').click();
+
+					cy.url().should('include', '/register/email');
+				},
+			);
+		});
+
+		it('existing users should fallback to the standard registration flow without passcodes', () => {
+			// Test users created via IDAPI-with-Okta do not have the activation
+			// lifecycle run at creation, so they don't transition immediately from
+			// STAGED to PROVISIONED (c.f.
+			// https://developer.okta.com/docs/reference/api/users/#create-user) .
+			// This is useful for us as we can test STAGED users first, then test
+			// PROVISIONED users in the next test by activating a STAGED user. Users
+			// created through Gateway-with-Okta do have this lifecycle run, so if we
+			// rebuild these tests to not use IDAPI at all, we need to figure out a
+			// way to test STAGED and PROVISIONED users (probably by just passing an
+			// optional `activate` prop to a createUser function).
```

Comment:
This comment doesn't really make sense here (I think it's copied from elsewhere).

Reply:
Good spot, I'll update this!